### ISSUE TITLE
[MDEPLOY-252] Deploy current archive without creating new archive (jar, war, ear)

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-deploy-plugin#540]
**[Ivica Mikic](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mindjoy)** opened **[MDEPLOY-252](https://issues.apache.org/jira/browse/MDEPLOY-252?redirect=false)** and commented

It would be great to be able to deploy current archive file (jar, war, ear) without creating a new one, to preserve timestamp from previous package or install run.

---

**Affects:** 2.8.2

**Remote Links:**
- [GitHub Pull Request #188
](https://github.com/apache/maven-deploy-plugin/issues/188)

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Why do you need maven-deploy-plugin. Based on a usual project which uses for example a maven-ear-plugin this will be deployed automatically via `mvn clean deploy` without supplemental configuration needed. Can you please give an example project and the commands, JDK version you are using and Maven version and how you triggered the build..

**Comment by jira-importer** [comment]
**[Ivica Mikic](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mindjoy)** commented

Here is my usual sequence of Maven build commands. First I run "mvn clean package" many times as I work on my code. Once my code is complete, I run "mvn install" then "mvn deploy", however, I noticed that Maven creates a new JAR, WAR or EAR files even though I don't have code changes and I don't specify "clean". I expect that Maven would just use previously created archived files (JAR, WAR, EAR) and "install" them to our local repository or "deploy" them to our global repository.

In a nutshell, I don't want to have different timestamps of the JAR, WAR, EAR files that I deployed on Java application servers after "mvn clean package" and those saved in Maven repositories after "mvn install" or "mvn deploy".

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

The first thing is to deploy servers only from a CI solution like Jenkins which means deploy the created artifacts into your repository manager and deploy your servers from there. Apart from that if you do `mvn install` and then `mvn deploy` you run several parts duplicate..cause the life cycle contains `install` . which means you only need to do `mvn clean deploy`. Furthermore if code has been changed can be identified by using your version control system...
The last sentence I don't understand. If you do `mvn clean deploy` all your artifacts have the same version etc. ?

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

The problem here is implied by the life cycle. If you call first `mvn clean package` and after that `mvn deploy` the life cycle starts from the beginning which includes `package` and other parts as well. This is related to incremental build which is currently not really working. So this means at the moment the files will be recreated. The question is: Are you doing your released locally ?

**Comment by jira-importer** [comment]
**[Ivica Mikic](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mindjoy)** commented

From the Maven perspective, we normally build our Java code first with "mvn clean package" many times during the active code change/development. This is done either manually on the command line or within the IDE. But once testing has passed and the code is considered reasonably stable, we want to proceed with the "mvn install", then "mvn deploy", but keep the original archive files in sync since there are no code changes. Isn't that reasonable? What would be the alternative?

I would still like to accomplish this somehow within the Maven life cycle, if possible.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

It sounds you like to work around the life cycle which sounds like fighting against Maven ...which usually is a combat you will loose ...Can you please make an example project let us discuss about this on a existing project which I think would make it easier...

**Comment by jira-importer** [comment]
**[Elliotte Rusty Harold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=elharo)** commented

project.buildTimestamp addresses this need in a different and more reproducible way
